nn-git-log/README.md

```diff
@@ -0,0 +1,83 @@
+
+To explain simply, the command `git log` and its corresponding optional flags lets you look at history of commits and how the link up in your respository. At the end of the chapter there will be links to additional resources, as well as more visual GUI tools you can use to look at the log of the respository, but I still suggest reading and learning this chapter.  
+
+This is because these tools still do rely on the git commands to generate their graphs, and understanding this command kind of gives you the 'feel' for it in general. Lastly, as this command comes with git, using the command does not require any additional setup other than having git.
+
+Note:
+This chapter serves to show you and explain the common use cases for the command. For a more detailed explaination, look at the addtional resources or the documentation of the command.
+
+## Project set up so-far
+
+Let's say the project you are working on is a full-stack web application, with a frontend and backend.  
+And you have decided to split the work to frontend and backend work, with respective features branches being worked on.
+
+It may look something like this:  
+```
+Github
+origin/master, master
+
+Teammate A
+frontend/landing
+
+Teammate B
+backend/start-server
+```
+
+## Looking at the small picture
+
+**Note: Assume you are teammate A**
+
+Let's say at this point you have added some code and commits to the project and want to see what commits you had done.  
+This could be because, you wanted to see where you left off yesterday or how features have or have not been done. (The later reason is when you are working on a larger feature)
+
+### Basic git log
+
+Well to look at all the commits made on your current branch you can do:
+```bash
+git log
+```
+![Basic git log of local branch](res/small_picture_1.png)
+
+From this screenshot you can see that although we can see all the commits linked to our current branch, there is a bit too much information at once.  
+For example information like when the commit was made or who made it is not really important when trying to look at the bigger picture.  
+This is also not practical when looking at projects with 10s or 100s of commits, as we would be lost in the information
+
+### Flag oneline
+
+Luckily there are special flags we can use to format the output.  
+If we add the flag, `--oneline`, git will shorten each commit to oneline, only preserving the important information.  
+Looking at the man page for this command:  
+```bash
+Documentation:
+--oneline
+This is a shorthand for "--pretty=oneline --abbrev-commit" used together.
+--abbrev-commit
+Instead of showing the full 40-byte hexadecimal commit object name, show only a partial prefix. Non default number of digits can be specified with "--abbrev=<n>" (which also modifies diff output, if it is displayed).
+```
+Thus when running the command.
+```bash
+git log --oneline
+```
+![Git log with only oneline](res/small_picture_2.png)
```

Comment:
Was wondering if adding the documentation here is good.
Documentation is taken from the man page.
[Man Page](https://git-scm.com/docs/git-log)

Honestly, I think it should not be included, but wanted to get your opinion on it @woojiahao

Reply:
I think that we can include the documentation for flags that are important to the discussion. For instance, in the commit above about `git log` have a simple description of `--oneline` does provide some help into showing ways to improve the use of the command. I don't think we need to include all of the possible flags. A link to the main page is sufficient for them to read more :thinking:

Reply:
Resolving this